functional-test/src/test/groovy/org/rundeck/util/common/jobs/JobUtils.groovy

```diff
@@ -116,26 +118,38 @@ class JobUtils {
         return jobs[0]
     }
 
+
     static Execution waitForExecutionToBe(
             String state,
             String executionId,
             ObjectMapper mapper,
             RdClient client,
-            int iterationGap,
-            int timeout
+            WaitingTime iterationGap,
+            WaitingTime timeout
+    ) {
+        return waitForExecutionToBe(state, executionId, mapper, client, iterationGap.duration, timeout.duration)
+    }
+
+    static Execution waitForExecutionToBe(
+            String state,
```

Comment:
I'd recommend making this a `List<String>` for flexibility.
A common use case is to wait for the job to "complete" -- either successfully or not -- represented by a multitude of statuses: `[         'aborted',             'failed',             'succeeded',            'timedout',             'other'])`

Reply:
Similar as the above comment. I like this idea, but its not in the scope of what i'm doing here. Let's make a different PR for this.

Reply:
I did implement it :D